### PR TITLE
Country note delete and country note edit fix

### LINKED ISSUE #283: [PE-D] Able to delete country notes which are not displayed
In the user guide, it is stated that:
> Deletes a note that is associated with a specific country at the given index based on the current view of the list panel displaying country notes.

However, it is possible to delete country notes which are not even being displayed. This is very confusing to the user.

Steps to replicate (using the default data on initial application load):
```
country note add c/SG nt/has one of the lowest corporate taxes in the world t/tax
country note delete 1
```

Full log:
```
INFO: =============================[ Initializing TBM ]=============================
Oct 30, 2020 4:41:52 PM seedu.address.MainApp start
INFO: Starting TbmManager V1.2.1ea
Oct 30, 2020 4:41:52 PM seedu.address.ui.UiManager start
INFO: Starting UI...
Oct 30, 2020 4:42:09 PM seedu.address.logic.LogicManager execute
INFO: ----------------[USER COMMAND][country note add c/SG nt/has one of the lowest corporate taxes in the world t/tax]
Oct 30, 2020 4:42:09 PM seedu.address.logic.parser.MainParser parseCommand
INFO: Command Type: country Rest of command:  note add c/SG nt/has one of the lowest corporate taxes in the world t/tax
Oct 30, 2020 4:42:09 PM seedu.address.logic.parser.MainParser parseCountryCommands
INFO: Command Type: note Rest of command:  add c/SG nt/has one of the lowest corporate taxes in the world t/tax
Oct 30, 2020 4:42:09 PM seedu.address.logic.parser.MainParser parseCountryNoteCommands
INFO: Command Type: add Rest of command:  c/SG nt/has one of the lowest corporate taxes in the world t/tax
Oct 30, 2020 4:42:09 PM seedu.address.model.note.Note <init>
INFO: --------------[New Note created with contents: has one of the lowest corporate taxes in the world]
Oct 30, 2020 4:42:09 PM seedu.address.model.note.Note <init>
INFO: --------------[New Note created with contents: ]
Oct 30, 2020 4:42:09 PM seedu.address.model.note.Note <init>
INFO: --------------[New Note created with contents: has one of the lowest corporate taxes in the world]
Oct 30, 2020 4:42:09 PM seedu.address.ui.MainWindow executeCommand
INFO: Result: Successfully added country note for Singapore (SG): [Singapore (SG)] has one of the lowest corporate taxes in the world
Oct 30, 2020 4:42:09 PM seedu.address.ui.MainWindow executeCommand
INFO: Widget View Option: NONE
Oct 30, 2020 4:42:14 PM seedu.address.logic.LogicManager execute
INFO: ----------------[USER COMMAND][country note delete 1]
Oct 30, 2020 4:42:14 PM seedu.address.logic.parser.MainParser parseCommand
INFO: Command Type: country Rest of command:  note delete 1
Oct 30, 2020 4:42:14 PM seedu.address.logic.parser.MainParser parseCountryCommands
INFO: Command Type: note Rest of command:  delete 1
Oct 30, 2020 4:42:14 PM seedu.address.logic.parser.MainParser parseCountryNoteCommands
INFO: Command Type: delete Rest of command:  1
Oct 30, 2020 4:42:14 PM seedu.address.ui.MainWindow executeCommand
INFO: Result: Deleted country note at index 1: [Singapore (SG)] has one of the lowest corporate taxes in the world
```

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: ianyong/ped#4

### PR DESCRIPTION
**Pull request by qwoprocks**
## Description

Refer to bug issue.

Fixes #283

## Testing

Ran all tests, passed.
Also tried deleting/editing country notes from a variety of views, for example default view, and client view, and country view. Works as expected.

## Remarks

NIL

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/CountryNoteEditCommand.java [hunk 1]
```diff
@@ -109,8 +112,4 @@ public boolean equals(Object other) {
         return targetIndex.equals(c.targetIndex) && countryNote.equals(c.countryNote) && tags.equals(c.tags);
     }
 
-    @Override
-    public int hashCode() {
```
Comment (raysonkoh) [inline src/main/java/seedu/address/logic/commands/CountryNoteEditCommand.java:113]: Why is this hashcode method deleted?

**Review by raysonkoh — CHANGES_REQUESTED** [review]
Did some manual testing, works as expected.

File: src/main/java/seedu/address/logic/commands/CountryNoteEditCommand.java (hunk 1 above)
Comment (qwoprocks) [reply src/main/java/seedu/address/logic/commands/CountryNoteEditCommand.java:113]: All the other command classes don't have the hashcode function, so I deleted this one to standardize

**Review by raysonkoh — APPROVED** [review]
LGTM

**Pull request merged**